### PR TITLE
QUARKUS-743 test plan

### PR DESCRIPTION
**Pull request by Sgitario**

### CODE REVIEW

File: QUARKUS-743.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ - Define the scope of the testing
+ - Analyze the possible impact on Quarkus itself - e.g performance, compatibility
+ - Determine impact on test suites - e.g. new test suite, increased complexity of existing test suite, maintenance burden
+
+ As part of the development, code starters are already being verified using `Maven` and `Gradle` in [these tests](https://github.com/quarkusio/quarkus/blob/master/integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartRunIT.java#L74-L90). 
+
+### Impact on testsuites and testing automation:
+ - Test development in Start-Stop testsuite in order to cover:
+   - Verify that the code start compile.
+   - Verify that the code start is fully working adding custom tests for each code start.
```
Comment (Sgitario) [inline QUARKUS-743.md:48]: Is this worthy? From my point of view, the code start test coverage included in integration-tests and in StartStop suite should verify this when running the application... 
If custom tests should be added, they should be part of the code starter.
What do you think?

File: QUARKUS-743.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+quarkus-qute
+quarkus-resteasy
+quarkus-resteasy-jackson
+quarkus-resteasy-reactive
+quarkus-spring-web
+```
+
+The list has been provided by running: `curl https://code.quarkus.io/api/extensions 2>/dev/null | jq '.[] | select(.tags[] | contains("provides-example")) | .id' | sort | uniq`.
+
+For each Quarkus extension with code starter:
```
Comment (rsvoboda) [inline QUARKUS-743.md:33]:
```suggestion
For each supported and tech preview Quarkus extension with code starter:
```

File: QUARKUS-743.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ - Verify that the Quarkus extensions defined in the `pom.xml` are marked with the `provides-example` tag in `code.quarkus.io`. 
+ - Verify that the code start compile.
+ - Verify that the code start is fully working.
+
+Following actions were taken to ensure test coverage and automation for native image:
+ - Define the scope of the testing
+ - Analyze the possible impact on Quarkus itself - e.g performance, compatibility
+ - Determine impact on test suites - e.g. new test suite, increased complexity of existing test suite, maintenance burden
+
+ As part of the development, code starters are already being verified using `Maven` and `Gradle` in [these tests](https://github.com/quarkusio/quarkus/blob/master/integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartRunIT.java#L74-L90).
```
Comment (rsvoboda) [inline QUARKUS-743.md:43]:
```suggestion
 As part of the development, code starters are already being verified using `Maven` in [these tests](https://github.com/quarkusio/quarkus/blob/master/integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartRunIT.java#L74-L90). 
```

File: QUARKUS-743.md (hunk 1 above)
Comment (rsvoboda) [reply QUARKUS-743.md:48]: +1

File: QUARKUS-743.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+Following actions were taken to ensure test coverage and automation for native image:
+ - Define the scope of the testing
+ - Analyze the possible impact on Quarkus itself - e.g performance, compatibility
+ - Determine impact on test suites - e.g. new test suite, increased complexity of existing test suite, maintenance burden
+
+ As part of the development, code starters are already being verified using `Maven` and `Gradle` in [these tests](https://github.com/quarkusio/quarkus/blob/master/integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartRunIT.java#L74-L90). 
+
+### Impact on testsuites and testing automation:
+ - Test development in Start-Stop testsuite in order to cover:
+   - Verify that the code start compile.
```
Comment (rsvoboda) [inline QUARKUS-743.md:47]: compiles || runs in dev mode || compiles + runs java -jar
Which one :) ?

**Review by rsvoboda — CHANGES_REQUESTED** [review]
please adjust the TP based on the comments

File: QUARKUS-743.md (hunk 1 above)
Comment (Sgitario) [reply QUARKUS-743.md]: I've removed this line then.

File: QUARKUS-743.md (hunk 4 above)
Comment (Sgitario) [reply QUARKUS-743.md]: I've changed this line to make it clear that it should work either DEV or JAV modes.

**Comment by Sgitario** [comment]
@rsvoboda I've updated the test plan based on your comments.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**